src/mixer/state.ts

```diff
@@ -498,7 +498,17 @@ export const load = (
           const itsChannel = getState()
             .showplan.plan!.filter((x) => x.channel === item.channel)
             .sort((x, y) => x.weight - y.weight);
-          const itsIndex = itsChannel.indexOf(item);
+          // Sadly, we can't just do .indexOf() item directly,
+          // since the player's idea of an item may be changed over it's lifecycle (setting intro/cue/outro etc.)
+          // Therefore we'll find the updated item from the plan and match that.
+          const itsChannelItem = itsChannel.filter(
+            (x) => itemId(x) === itemId(item)
+          );
+          if (itsChannelItem.length !== 1) {
+            // Somehow we've got 0 or multiple identical timeslotitems (or ghosts), bail out!
+            return;
+          }
+          const itsIndex = itsChannel.indexOf(itsChannelItem[0]);
```

Comment:
Have you considered `itsChannel.findIndex(x => itemId(x) === itemId(item))`?

Reply:
I have now 👍